File: src/main/java/com/prmncr/normativecontrol/dtos/Result.java
package com.prmncr.normativecontrol.dtos;

import lombok.Getter;

import java.util.List;

public class Result {
    @Getter
    private final boolean isFail;
    @Getter
    private final FailureType failureType;
    @Getter
    private final List<Error> errors;

    public Result(FailureType failureType) {
        this.isFail = true;
        this.failureType = failureType;
        errors = null;
    }

    public Result(List<Error> errors) {
        this.isFail = false;
        this.failureType = FailureType.NONE;
        this.errors = errors;
    }
}

File: src/main/kotlin/com/prmncr/normativecontrol/dtos/Result.kt
package com.prmncr.normativecontrol.dtos

class Result {
    private val isFail: Boolean
    private val failureType: FailureType
    val errors: List<Error>

    constructor(failureType: FailureType) {
        isFail = true
        this.failureType = failureType
        errors = listOf()
    }

    constructor(errors: List<Error>) {
        isFail = false
        failureType = FailureType.NONE
        this.errors = errors
    }
}